app/bundles/EmailBundle/MonitoredEmail/Mailbox.php

```diff
@@ -393,7 +393,12 @@ protected function initImapStream()
      */
     protected function isConnected(): bool
     {
-        return $this->isConfigured() && $this->imapStream && @imap_ping($this->imapStream);
+        if (function_exists('imap_open')) {
+            // imap_is_open is only available from PHP 8.2.1
+            return $this->isConfigured() && $this->imapStream && imap_is_open($this->imapStream);
+        } else {
+            return $this->isConfigured() && $this->imapStream && @imap_ping($this->imapStream);
```

Comment:
```suggestion
            return $this->isConfigured() && $this->imapStream && is_resource($this->imapStream) && @imap_ping($this->imapStream);
```
Let's use this logic as a fallback, since it was reported that reverting the change from Mautic 5.1.1 fixed this bug.

For reference, this is from Mautic 5.1.1:
https://github.com/mautic/mautic/blob/8fa5a716b3082edb6bf3f2c10ab8025729d95a9c/app/bundles/EmailBundle/MonitoredEmail/Mailbox.php#L395